web/courses_test.go

```diff
@@ -82,10 +82,9 @@ func TestGetCourses(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-
 	for i, course := range foundCourses.Courses {
-		if !reflect.DeepEqual(course, testCourses[i]) {
-			t.Errorf("have course %+v want %+v", course, testCourses[i])
+		if diff := cmp.Diff(testCourses[i], course, protocmp.Transform()); diff != "" {
```

Comment:
Here it is possible to drop the for-loop and compare all with a single `cmp.Diff`.
```go
	gotCourses := foundCourses.Courses
	if diff := cmp.Diff(wantCourses, gotCourses, protocmp.Transform()); diff != "" {
		t.Errorf("TestGetCourses() mismatch (-wantCourses +gotCourses):\n%s", diff)
	}
```

Also, please use the function name being tested in the string: `GetCourses()` instead of `TestGetCourse`.

And rename `testCourses` to `wantCourses`. Remember you can safely rename things in VSCode by selecting the variable, type, function text and clicking F2 and give the new name...